### DISCUSSION TITLE
How to raise an error from non-invoke?

### DISCUSSION THREAD

**Discussion by thetumper** [statelyai/xstate#4888] [Q&A]
In response to an on "child.error" event, from multiple children, I need to accumulate possibly multiple errors.  The parent state receives the "child.error" event and concats each error to context.errors (array).  But as the parent state is not an invoke, I cannot figure out how to raise or throw an error, and specify handling in an onError block?

This does not help, since it describes only the invoke scenario:
https://stately.ai/docs/migration#actions-can-throw-errors-without-escalate

I must be missing something?  Any ideas?

**Comment by davidkpiano** [comment]
There is no state-wide `onError` yet, but it's a good idea. You would have to handle this by sending "error" events and handling those just like any other event, for now.

If you could share some example code, maybe I can demonstrate how you could do this.

**Reply by thetumper** [reply]
Thanks for the response!  I did get this working with events.  In my case, I had a bit of struggle with a machine that is sometimes spawned as a child, and sometimes run from an actor (unit tests).  I had to modify the event sending, to not use sendParent for the latter.  I tried sendTo, with the system.get(parentId), but it didn't seem to work.  Passing in a full ref of the parent, and then checking if that exists before doing parent.send (not doing when parent is not there, and only having the parent ref in the spawn case), worked for me.

This steered me to the "parent.send" option:
https://github.com/statelyai/xstate/discussions/4838

Thanks again!
